plugins/modules/ndo_tenant_span_session.py

```diff
@@ -0,0 +1,587 @@
+            )
+
+        if match:
+
+            mso_update_values = {
+                "name": name,
+                "description": description,
+                ("destination", "mtu"): mtu,
+            }
+            source_group = mso_values.get("sourceGroup")
+            if source_group:
+                mso_update_values[("sourceGroup", "enableAdminState")] = source_group.get("enableAdminState")
+                mso_update_values[("sourceGroup", "sources")] = source_group.get("sources")
+            if destination_epg:
+                remote_group = mso_values.get("destination", {}).get("remote")
+                if remote_group:
+                    mso_update_values[("destination", "remote", "epgRef")] = remote_group.get("epgRef")
+                    mso_update_values[("destination", "remote", "spanVersion")] = remote_group.get("spanVersion")
+                    mso_update_values[("destination", "remote", "enforceSpanVersion")] = remote_group.get("enforceSpanVersion")
+                    mso_update_values[("destination", "remote", "destIPAddress")] = remote_group.get("destIPAddress")
+                    mso_update_values[("destination", "remote", "srcIPPrefix")] = remote_group.get("srcIPPrefix")
+                    mso_update_values[("destination", "remote", "flowID")] = remote_group.get("flowID")
+                    mso_update_values[("destination", "remote", "ttl")] = remote_group.get("ttl")
+                    mso_update_values[("destination", "remote", "dscp")] = remote_group.get("dscp")
+
+            proposed_payload = copy.deepcopy(match.details)
+            append_update_ops_data(ops, proposed_payload, span_session_path, mso_update_values)
+            mso.sanitize(proposed_payload, collate=True)
+        else:
+            mso.sanitize(mso_values)
+            ops.append(dict(op="add", path="{0}/-".format(path), value=mso.sent))
+
+    elif state == "absent":
+        if match:
+            ops.append(dict(op="remove", path=span_session_path))
+
+    if not module.check_mode and ops:
+        response = mso.request(mso_template.template_path, method="PATCH", data=ops)
+        node_profiles = response.get("monitoringTemplate", {}).get("template", {}).get("spanSessions") or []
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            node_profiles,
+            [KVPair("uuid", uuid) if uuid else KVPair("name", name)],
+        )
+        if match:
+            set_template_and_references(mso_template, match.details)
+            mso.existing = match.details  # When the state is present
+        else:
+            mso.existing = {}  # When the state is absent
+    elif module.check_mode and state != "query":  # When the state is present/absent with check mode
+        set_template_and_references(mso_template, mso.proposed)
+        mso.existing = mso.proposed if state == "present" else {}
+
+    mso.exit_json()
+
+
+def set_template_and_references(mso_template, config):
```

Comment:
This function is likely not required if we use the updated update_config_with_template_and_references() in https://github.com/CiscoDevNet/ansible-mso/pull/632/files which recursively updates the config with references..